packages/sign-client/src/controllers/engine.ts

```diff
@@ -826,25 +822,17 @@ export class Engine extends IEngine {
     const { id, params } = payload;
     try {
       this.isValidSessionSettleRequest(params);
-      const {
-        relay,
-        controller,
-        expiry,
-        namespaces,
-        requiredNamespaces,
-        optionalNamespaces,
-        sessionProperties,
-        pairingTopic,
-      } = payload.params;
+      const { relay, controller, expiry, namespaces, sessionProperties, pairingTopic } =
+        payload.params;
       const session = {
         topic,
         relay,
         expiry,
         namespaces,
         acknowledged: true,
         pairingTopic,
-        requiredNamespaces,
-        optionalNamespaces,
+        requiredNamespaces: {},
```

Comment:
Why set to empty object here? Guessing simply to satisfy types since it looks like these keys are overwritten with `proposal.[key]` anyways on the engineEvent handler above?

Reply:
yes, thought its redundant to create new `connect` event type just for this